src/main/java/com/serotonin/bacnet4j/obj/TrendLogObject.java

```diff
@@ -204,10 +205,35 @@ public boolean isLogDisabled() {
         return logDisabled;
     }
 
+    /**
+     * @deprecated This method return a buffer that may not be thread-safe. Use {@link #doWithBuffer} instead.
+     * This method will be removed in a future major version.
+     */
+    @Deprecated
     public LogBuffer<LogRecord> getBuffer() {
         return buffer;
     }
 
+    public void doWithBuffer(Consumer<LogBuffer<LogRecord>> consumer) {
+        synchronized (buffer) {
+            consumer.accept(buffer);
+        }
+    }
+
+    public int getRecordCount() {
+        // Synchronize the buffer before requesting the size because we don't know the implementation of the buffer,
+        // and whether the operation is atomic or not.
+        synchronized (buffer) {
+            return buffer.size();
+        }
+    }
+
+    public LogRecord getRecord(int index) {
+        synchronized (buffer) {
+            return buffer.get(index);
+        }
+    }

```

Comment:
Would it be possible to synchronize the methods in `LinkedListLogBuffer` directly, rather than synchronizing individually on each object?

Reply:
Yeah, that would that class thread-safe (across individual method calls at least), but the thing is the trend logs allow the code to provide any implementation of LogBuffer, not just the one provided by the library. So, for example, someone could write a disk-based buffer implementation, which would not necessarily be thread safe. (Although arguably this would be their own problem.)